### PR TITLE
chore: adding support of PUT method

### PR DESCRIPTION
**Pull request by sbansla**
# Fixes #
Making http methods operation independent. Now any method can be used for any operation as defined in open api spec.
I have tested all the helper libs and terraform by generating code from open api specs (twilio-oai rc branch), It is working as expected.

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [ ] Run `make test-docker`
- [ ] Verify affected language:
    - [ ] Generate [twilio-go](https://github.com/twilio/twilio-go) from our [OpenAPI specification](https://github.com/twilio/twilio-oai) using the [build_twilio_go.py](./examples/build_twilio_go.py) using `python examples/build_twilio_go.py path/to/twilio-oai/spec/yaml path/to/twilio-go` and inspect the diff
    - [ ] Run `make test` in `twilio-go`
    - [ ] Create a pull request in `twilio-go`
    - [ ] Provide a link below to the pull request
- [ ] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [ ] I have read the [Contribution Guidelines](https://github.com/twilio/twilio-oai-generator/blob/main/CONTRIBUTING.md) and my PR follows them
- [ ] I have titled the PR appropriately
- [ ] I have updated my branch with the main branch
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation about the functionality in the appropriate .md file
- [ ] I have added inline documentation to the code I modified

If you have questions, please create a GitHub Issue in this repository.

### CODE REVIEW

**Comment by AsabuHere** [comment]
Since PUT method have payload, ValidationClient in java sdk needs to be modified

File: src/main/java/com/twilio/oai/api/JavaApiResourceBuilder.java [hunk 1]
```diff
@@ -15,11 +16,32 @@
...
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.TreeSet;
```
[the remark this answers is not in the document]
Comment (sbansla) [reply src/main/java/com/twilio/oai/api/JavaApiResourceBuilder.java:30]: Ideally we import only those classes which we want to use.

File: gen.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import argparse
```
[the remark this answers is not in the document]
Comment (sbansla) [reply gen.py:1]: Let me remove file, This was added by mistake

File: src/main/java/com/twilio/oai/api/ApiResourceBuilder.java [hunk 3]
```diff
@@ -91,6 +92,8 @@ public ApiResourceBuilder updateOperations(Resolver<CodegenParameter> codegenPar
         return this;
     }
 
+    public abstract void updateHttpMethod(CodegenOperation co);
```
[the remark this answers is not in the document]
Comment (sbansla) [reply src/main/java/com/twilio/oai/api/ApiResourceBuilder.java]: resolved

**Comment by sbansla** [comment]
> Since PUT method have payload, ValidationClient in java sdk needs to be modified

done.

**Review by tiwarishubham635 — APPROVED** [review]

**Pull request merged**